python_modules/dagster/dagster_tests/core_tests/definitions_tests/decorators_tests/test_op.py

```diff
@@ -778,3 +778,64 @@ def the_graph_provides_inputs():
 
     result = the_graph_provides_inputs.execute_in_process()
     assert result.success
+
+
+def test_generic_output_op():
+    @op
+    def the_op() -> Output[str]:
+        return Output("foo")
+
+    assert the_op.output_def_named("result").dagster_type.key == "String"
+
+    result = execute_op_in_graph(the_op)
+    assert result.success
+    assert result.output_for_node("the_op") == "foo"
+
+    @op
+    def the_op_bad_type_match() -> Output[int]:
+        return Output("foo")
+
+    with pytest.raises(
+        DagsterTypeCheckDidNotPass,
+        match='Type check failed for step output "result" - expected type '
+        '"Int". Description: Value "foo" of python type "str" must be a int.',
+    ):
+        execute_op_in_graph(the_op_bad_type_match)
+
+
+def test_output_generic_correct_inner_type():
+    @op
+    def the_op_not_using_output() -> Output[int]:
+        return 42
+
+    result = execute_op_in_graph(the_op_not_using_output)
+    assert result.success
```

Comment:
want to make sure we're ok with this; the alternative would be erroring unless someone explicitly provided an output.

Reply:
I think its fine for us to be flexible here

we should add a test case for type mismatch here though for clarity